runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1017,6 +1052,99 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 #endif /* J9VM_ENV_DATA64 */
 	}
 
+	/**
+	 * Extract the class pointer from an unforwarded object.
+	 *
+	 * This method will assert if the object has been marked as forwarded.
+	 *
+	 * @param[in] pointer to forwardedHeader the MM_ForwardedHeader instance encapsulating the object
+	 * @return pointer to the J9Class from the object encapsulated by forwardedHeader
+	 * @see MM_ForwardingHeader::isForwardedObject()
+	 */
+	MMINLINE J9Class *
+	getPreservedClass(MM_ForwardedHeader *forwardedHeader)
```

Comment:
This method actually needs 0 knowledge from (indexable)objectModel. It could just be a method in MM_ForwardedHeader. But then it has J9 specific knowledge. So we could have language speicific MM_ForwardedHeaderDelegate and have it there (and remove from various ObjectModels and thus avoid clones, and avoid circular dependancies...)

Other preserve type of methods could stay in ObjectModel since they do need some knowledge (various shape of headers and how to determine the size etc.)

Just a rant. Nothing to do at this round of changes.